### ISSUE TITLE
打分结果不稳定

### ISSUE THREAD

**Issue by xuqiang** [zhihu/cuBERT#21]
打分结果不稳定，可以稳定复现。追了下代码 ，和bert.cpp中 buffer_filled有关，删除buffer_filled相关逻辑之后，结果稳定。

请教下 buffer_filled当时设计之初的作用是什么？

https://github.com/zhihu/cuBERT/blob/master/src/cuBERT/Bert.cpp#L103

**Comment by levyfan** [comment]
计算过程中，有一部分 copy 操作不依赖于输入，可以在输入到来之前执行，或者在返回输出结果之后，异步地执行，不占用 latency 时间。这就是 _pre_compute 那些函数做的事情。

但是如果在 compute 完了之后，多次调用 logits/pooled_output/sequence_output 那些方法，就会有 buffer 被误清洗的问题。所以目前设计上，是一次 compute 对应一次 output，如果一次 compute 想要对应多次 output，就需要改下代码。

**Comment by levyfan** [comment]
这部分 copy 操作主要包括 fc 中将 bias 提前拷贝到 output_buffer 内，这样只需要执行一次 gemm 就可以得到 output = w * input + bias 的结果。

**Comment by levyfan** [comment]
另外，这里面的异步其实是由 CUDA 本身实现的，当我们执行 _pre_compute 时，代码会将 gpu 操作发送过去，然后直接返回，不会等待执行结果结束。这样就缩短了响应时间。

**Comment by xuqiang** [comment]
了解 多谢 

我看看改下这块逻辑

**Comment by levyfan** [comment]
发现 bug 是由于 Bert.cpp 几处没有调用 `additional_output_layer._pre_compute` 引起，已经在 https://github.com/zhihu/cuBERT/pull/24 里修复
